src/opensearch.py

```diff
@@ -35,21 +35,61 @@
 logger = logging.getLogger(__name__)
 
 
+BASE_SNAP_DIR = "/var/snap/opensearch"
+SNAP_DATA = f"{BASE_SNAP_DIR}/current"
+SNAP_COMMON = f"{BASE_SNAP_DIR}/common"
+SNAP = "/snap/opensearch/current"
+
+
+OPENSEARCH_SNAP_METADATA = {
+    40: {
+        "paths": Paths(
+            home=f"{SNAP_DATA}/usr/share/opensearch",
+            conf=f"{SNAP_DATA}/etc/opensearch",
+            data=f"{SNAP_COMMON}/var/lib/opensearch",
+            logs=f"{SNAP_COMMON}/var/log/opensearch",
+            jdk=f"{SNAP}/usr/lib/jvm/java-21-openjdk-amd64",
+            tmp=f"{SNAP_COMMON}/usr/share/tmp",
+            bin=f"{SNAP}/usr/share/opensearch/bin",
+        ),
+    },
+    0: {
+        "paths": Paths(
+            home=f"{SNAP_DATA}/usr/share/opensearch",
+            conf=f"{SNAP_DATA}/etc/opensearch",
+            data=f"{SNAP_COMMON}/var/lib/opensearch",
+            logs=f"{SNAP_COMMON}/var/log/opensearch",
+            jdk=f"{SNAP}/usr/lib/jvm/java-17-openjdk-amd64",
+            tmp=f"{SNAP_COMMON}/usr/share/tmp",
+            bin=f"{SNAP}/usr/share/opensearch/bin",
+        ),
+    },
+}
+
+
 class OpenSearchSnap(OpenSearchDistribution):
     """Snap distribution of opensearch, only overrides properties and logic proper to the snap."""
 
-    _BASE_SNAP_DIR = "/var/snap/opensearch"
-    _SNAP_DATA = f"{_BASE_SNAP_DIR}/current"
-    _SNAP_COMMON = f"{_BASE_SNAP_DIR}/common"
-    _SNAP = "/snap/opensearch/current"
+    _BASE_SNAP_DIR = BASE_SNAP_DIR
+    _SNAP_DATA = SNAP_DATA
+    _SNAP_COMMON = SNAP_COMMON
+    _SNAP = SNAP
 
     def __init__(self, charm, peer_relation: str):
+        for attempt in Retrying(stop=stop_after_attempt(5), wait=wait_fixed(wait=5)):
```

Comment:
why this before the `super` call?

Reply:
Because `super().__init__()` invokes:
```
        self.paths = self._build_paths()
        self._set_env_variables()
```

And both, in `src/opensearch` need the a pre-defined `self._opensearch`.